main.go

```diff
@@ -436,13 +437,45 @@ func setupReconcilers(ctx context.Context, mgr ctrl.Manager) webhooks.ClusterCac
 		})
 	}
 
+	// Setup a separate cache for the metadata watches to secrets.
+	// This way the watch does not use the LabelSelector defined at the cache which
+	// would filter to secrets having the cluster label, because secrets referred
+	// by ClusterResourceSet or ExtensionConfig are not specific to a single cluster.
+	partialSecretCache, err := cache.New(mgr.GetConfig(), cache.Options{
+		Scheme:            mgr.GetScheme(),
+		Mapper:            mgr.GetRESTMapper(),
+		HTTPClient:        mgr.GetHTTPClient(),
+		SyncPeriod:        syncPeriod,
+		DefaultNamespaces: watchNamespaces,
+		DefaultTransform: func(in interface{}) (interface{}, error) {
+			// Use DefaultTransform to drop objects we don't expect to get into this cache.
+			obj, ok := in.(*metav1.PartialObjectMetadata)
+			if !ok {
+				return nil, fmt.Errorf("cache expected to only get PartialObjectMetadata, got %T", in)
```

Comment:
What happens if we hit this or the next case? (does the controller fail? are we getting not founds on get? anything else?